frontends/smvparser.y

```diff
@@ -437,16 +663,20 @@ simple_expr: constant {
                 smt::Term e = enc.solver_->make_term(smt::BVOr, a->getTerm(), b->getTerm());
```

Comment:
So it looks like all of these operators are using `BV*` ops, when I think they should be boolean ops. Or, you might actually need to check the sorts and use the appropriate operator if they're overloaded (meaning it uses the same symbol in SMV for Or over booleans and bit-vectors). If it is overloaded, then you can do the same thing you did for arithmetic operators, i.e.

```
Term at = a->getTerm();
Term bt = b->getTerm();
SortKind ask = a->get_sort()->get_sort_kind();
SortKind bsk = b->get_sort()->get_sort_kind();
if (at->get_sort() != bt->get_sort() || (ask != BOOL && ask != BV))
{
  throw PonoException("Expecting two booleans or two bit-vectors of the same width");
}

assert(ask == bsk);
assert(ask == BOOL || ask == BV);
Term e;
if (ask == BOOL)
{
   e = enc.solver_->make_temr(smt::Or, at, bt);
}
else
{
   e = enc.solver_->make_term(smt::BVOr, at, bt);
}

assert(e); // check that e is non-null
$$ = new SMVnode(e, bvs_a);
```